Stargazer

HelenMamontova opened the pull request:
Plugin Radius added.
Parsing command line parameters added.
Implemented starting Stargazer without starting daemon.

HelenMamontova committed:
Radius. Block if (BUILD_MOD_RADIUS) added.

HelenMamontova committed:
Radius. Catalog radius with radius.h and radius.cpp files added.

HelenMamontova committed:
Declaration of methods GetVersion, SendMessage, SetUsers, Start added.

HelenMamontova committed:
Definition of method SendMessage added.

HelenMamontova committed:
The option for BUILD_MOD_RADIUS added.

HelenMamontova committed:
Header files <string>, <vector>, <list> added. Virtual methods Start,
Stop declaration added. Private methods AddUser, DelUser, GetUser,
UpdateUserAuthorization added. Class member variable errorStr, users,
userList, isRunning, m_conns, m_onAddUserConn, m_onDelUserConn added.

HelenMamontova committed:
Header files <functional>, "stg/user.h" added. Overriding virtual
methods Start, Stop added.

HelenMamontova committed:
Private methods SetUserNotifiers, UnSetUserNotifiers declaration added.

HelenMamontova committed:
Header file <cassert> added. Definition methods GetUser, AddUser, DelUser,
UpdateUserAuthorization added.

HelenMamontova committed:
Unnecessary block private removed. Unnecessary header files removed.

HelenMamontova committed:
Unnecessary methods removed. Unnecessary header files removed.

HelenMamontova committed:
Overriding virtual methods Start, Stop, Reload, IsRunning, ParseSettings, GetStrErroor,
GetStartPosition, GetStopPosition, SendMessage added. Class member
variable errorStr, isRunning added.

HelenMamontova committed:
Definition of methods Start, Stop, SendMessage removed.

HelenMamontova committed:
Header file <string> added. The functions PrintHelp, PrintVersion added.
Parsing command line parameters added. The parameter of function settings
when it call changed.

HelenMamontova committed:
Radius. Variable bool noDaemon added. Parameter -f/--f added to parsing
command line. The call of function ForkAndWait changed - it is called if the variable
noDaemon is false.

madf commented on projects/stargazer/main.cpp:
```diff
@@ -208,6 +209,21 @@ void KillExecuters()
+
+void PrintHelp(const std::string& programName)
+{
+    std::cout << "Usage: " << programName << "[-h/--help] [-v/--version] [-f/--foreground] [../dist]\n"
```
`[../dist]` -> `<conf-dir-path>`

madf commented on projects/stargazer/main.cpp:
```diff
@@ -208,6 +209,21 @@ void KillExecuters()
+              << "\t --help, -h            - print this help;\n"
+              << "\t --version, -v         - print version;\n"
+              << "\t --foreground, -f      - do not go into background;\n"
+              << "\t ../dist               - path to the directory where the configuration file is located .\n";
```
See above

madf commented on projects/stargazer/main.cpp:
```diff
@@ -223,7 +239,33 @@ int main(int argc, char* argv[])
+            PrintVersion(argv[0]);
+            return 0;
+        }
+        if (arg == "../dist")
```
This is plain wrong

madf commented on projects/stargazer/plugins/other/radius/radius.cpp:
```diff
@@ -0,0 +1,19 @@
+
+std::string RADIUS::GetVersion() const
+{
+    return "Radius authorizator v.1.0";
```
Please remove 'authorizator'. This is not an authorization plugin, although it can optionally perform authorization. Just 'Radius' is fine.

HelenMamontova committed:
Radius. Command line parameter 'path' value fixed in the function
PrintHelp. Parsing command line parameters fixed. The key
-f/--foreground added to parsing command line parameters.

HelenMamontova committed:
Radius. Return value in the function GetVersion changed.

madf commented on projects/stargazer/main.cpp:
```diff
@@ -208,6 +209,21 @@ void KillExecuters()
+              << "\t --help, -h            - print this help;\n"
+              << "\t --version, -v         - print version;\n"
+              << "\t --foreground, -f      - do not go into background;\n"
+              << "\t <conf-dir-path>       - path to the directory where the configuration file is located .\n";
```
Extra whitespace before '.'.

madf commented on projects/stargazer/main.cpp:
```diff
@@ -223,7 +239,39 @@ int main(int argc, char* argv[])
+            {
+                noDaemon = true;
+
+                if (argc == i + 2)
```
Why do you need this condition? Wouldn't the following command line argument go into the 'else' branch on the next cycle iteration?

HelenMamontova committed:
Radius. Unnecessary check 'argc = i + 2' removed from parsing command
line parameters.

madf reviewed (changes requested):
I am okay with your changes. However, there is a macro out there: NO_DAEMON. It used to control the way we start stargazer (whether to make it a daemon or not). Now, with your changes, we control it in runtime by command line arguments. We need to adjust the logic driven by the macro to use command line argument, and completely eliminate the macro.

HelenMamontova committed:
The macro NO_DAEMON removed. The logic driven by the macro changed to
command line argument f analysis.

madf commented on projects/stargazer/main.cpp:
```diff
@@ -367,9 +356,7 @@ int main(int argc, char* argv[])
     WriteServLog("+++++++++++++++++++++++++++++++++++++++++++++");
 
-#ifndef NO_DAEMON
     creat(startFile.c_str(), S_IRUSR);
```
Do we still need to create the file in no-daemon mode?

madf commented on projects/stargazer/main.cpp:
```diff
@@ -279,9 +270,7 @@ int main(int argc, char* argv[])
     }
 
-#ifndef NO_DAEMON
     const auto startFile = settings.GetConfDir() + START_FILE;
```
It seems that `startFile` is only used in `creat`. It makes sense to move it closer to the point of use.

HelenMamontova committed:
Radius. The check variable noDaemon added when create function call.

madf commented on projects/stargazer/main.cpp:
```diff
@@ -356,7 +354,10 @@ int main(int argc, char* argv[])
     WriteServLog("+++++++++++++++++++++++++++++++++++++++++++++");
 
-    creat(startFile.c_str(), S_IRUSR);
+    const auto startFile = settings.GetConfDir() + START_FILE;
```
And if `noDaemon` is `true`, we do not need this variable either.

madf commented on projects/stargazer/main.cpp:
```diff
@@ -223,7 +230,34 @@ int main(int argc, char* argv[])
 
-    SettingsImpl settings(argc == 2 ? argv[1] : "");
+    std::string path;
+    bool noDaemon(false);
```
Consider `bool noDaemon = false;`, it looks more naturally.

HelenMamontova committed:
Radius. The variable startFile definition moved to block if(!noDaemon)
in main function.
